### DISCUSSION TITLE
`withDrafts` on the `Route::`

### DISCUSSION THREAD

**Discussion by SergkeiM** [oddvalue/laravel-drafts#18] [Ideas]
Laravel has a method `withTrashed` on `Route::` [Soft Deleted Models](https://laravel.com/docs/10.x/routing#implicit-soft-deleted-models)

Currently [laravel-drafts](https://github.com/oddvalue/laravel-drafts) default behavior is to fetch `withoutDrafts()`,

A workaround is to add [Customizing The Resolution Logic](https://laravel.com/docs/10.x/routing#customizing-the-resolution-logic):

```php
Route::bind('draft', fn (string $value) => Page::withDrafts()->findOrFail($value));
```

I think it would be nice to have `withDrafts` on Route:

```php
Route::post('drafts/{draft}/publish', [DraftController::class, 'publish'])->name('drafts.publish')->withDrafts();
```

Currently I'm digging throug Laravel source code to see their `withTrashed()` implementation, if I find a way to similar for `withDrafts` will create a PR.

**Comment by oddvalue** [comment]
@Froxz this feature has been added to v1.3.0
https://github.com/oddvalue/laravel-drafts/tree/v1.3.0#withdraftsmiddleware
